x-pack/legacy/plugins/siem/public/pages/case/components/tag_list/index.tsx

```diff
@@ -43,37 +37,35 @@ const MyFlexGroup = styled(EuiFlexGroup)`
   `}
 `;
 
-export const TagList = React.memo(({ tags, isEditTags, iconAction }: TagListProps) => {
+export const TagList = React.memo(({ isLoading, onSubmit, tags }: TagListProps) => {
   const { form } = useForm({
     defaultValue: { tags },
     options: { stripEmptyFields: false },
     schema,
   });
+  const [isEditTags, setIsEditTags] = useState(false);
 
-  const onSubmit = useCallback(async () => {
+  const onSubmitTags = useCallback(async () => {
     const { isValid, data: newData } = await form.submit();
-    if (isValid && iconAction) {
-      iconAction.onSubmit(newData.tags);
-      iconAction.onClick(false);
+    if (isValid && newData.tags) {
+      onSubmit(newData.tags);
+      setIsEditTags(false);
     }
   }, [form]);
```

Comment:
you need to add `onSubmit` since it is coming as a prop
```suggestion
  }, [form, onSubmit]);
```